crates/l2/proposer/l1_committer.rs

```diff
@@ -0,0 +1,502 @@
+use std::{collections::HashMap, time::Duration};
+
+use crate::{
+    proposer::state_diff::{AccountStateDiff, DepositLog, WithdrawalLog},
+    utils::{
+        config::{committer::CommitterConfig, eth::EthConfig},
+        eth_client::{transaction::blob_from_bytes, EthClient},
+        merkle_tree::merkelize,
+    },
+};
+use bytes::Bytes;
+use c_kzg::{Bytes48, KzgSettings};
+use ethereum_rust_blockchain::constants::TX_GAS_COST;
+use ethereum_rust_core::{
+    types::{
+        BlobsBundle, Block, EIP1559Transaction, GenericTransaction, PrivilegedL2Transaction,
+        PrivilegedTxType, Transaction, TxKind, BYTES_PER_BLOB,
+    },
+    Address, H256, U256,
+};
+use ethereum_rust_rpc::types::transaction::WrappedEIP4844Transaction;
+use ethereum_rust_storage::Store;
+use ethereum_rust_vm::{evm_state, execute_block, get_state_transitions};
+use keccak_hash::keccak;
+use secp256k1::SecretKey;
+use sha2::{Digest, Sha256};
+use std::ops::Div;
+use tokio::time::sleep;
+use tracing::{error, info, warn};
+
+use super::{errors::CommitterError, state_diff::StateDiff};
+use crate::utils::eth_client::{errors::EthClientError, eth_sender::Overrides};
```

Comment:
As an addition to the following suggestion (didn't make it bc it'd involve formatting):
- Merge the `crate` imports.
- Use `crate` instead of `super`.
```suggestion
use std::{collections::HashMap, time::Duration};
use crate::{
    proposer::state_diff::{AccountStateDiff, DepositLog, WithdrawalLog},
    utils::{
        config::{committer::CommitterConfig, eth::EthConfig},
        eth_client::{transaction::blob_from_bytes, EthClient},
        merkle_tree::merkelize,
    },
};
use bytes::Bytes;
use c_kzg::{Bytes48, KzgSettings};
use ethereum_rust_blockchain::constants::TX_GAS_COST;
use ethereum_rust_core::{
    types::{
        BlobsBundle, Block, EIP1559Transaction, GenericTransaction, PrivilegedL2Transaction,
        PrivilegedTxType, Transaction, TxKind, BYTES_PER_BLOB,
    },
    Address, H256, U256,
};
use ethereum_rust_rpc::types::transaction::WrappedEIP4844Transaction;
use ethereum_rust_storage::Store;
use ethereum_rust_vm::{evm_state, execute_block, get_state_transitions};
use keccak_hash::keccak;
use secp256k1::SecretKey;
use sha2::{Digest, Sha256};
use std::ops::Div;
use tokio::time::sleep;
use tracing::{error, info, warn};
use super::{errors::CommitterError, state_diff::StateDiff};
use crate::utils::eth_client::{errors::EthClientError, eth_sender::Overrides};
```

Reply:
done